qutebrowser/utils/qtutils.py

```diff
@@ -86,6 +85,13 @@ def _init_filedev(self, dev: QFileDevice, msg: str) -> str:
         msg += ": {!r}".format(filename)
         return msg
 
+def parse_version(version):
+    """Parse version string.
+
+    This is a replacement of pkg_resources.parse_version
+    """
+    v_q, _suffix = QVersionNumber.fromString(version)
+    return v_q.normalized()
```

Comment:
I don't think the `parse_version` function should be here, as `qtutils` is for Qt-specific utilities. Wouldn't it fit much better in `utils/version.py`? I guess you could even just call it `parse` then, as `version.parse(...)` seems clear enough.

Reply:
Thank you for the suggestion!

Btw, what's the correct way to annotate the type info, `def parse_version(version: str) -> QVersionNumber:` still fails mypy test,
```
qutebrowser/utils/qtutils.py:130: error: Unsupported left operand type for > ("QVersionNumber")  [operator]
        return (parse_version(qWebKitVersion()) >
                ^
```

Reply:
Huh, that seems like a bug in the type stubs. Feel free to annotate it as `QVersionNumber` and I'll take care of adding a workaround for it which doesn't involve sprinking `type: ignore[operator]` everywhere it's used.

Reply:
> I don't think the `parse_version` function should be here, as `qtutils` is for Qt-specific utilities. Wouldn't it fit much better in `utils/version.py`? I guess you could even just call it `parse` then, as `version.parse(...)` seems clear enough.

If we put it in `util/version.py`, it would lead to a circular import. `qtutils.version_check` would import `version.parse` and version would import `pdfjs` which in turn would import `qtutils`.

Reply:
@The-Compiler How about we moving `qtutils.version_check` to `version.check`? Then pdfjs doesn't need to import `qtuitls`.

Reply:
> @The-Compiler How about we moving `qtutils.version_check` to `version.check`? Then pdfjs doesn't need to import `qtuitls`.

Oh, this doesn't resolve the issue because of `qtutils.version_check -> version.parse -> pdfjs -> jinja -> qtutils`.

Reply:
I'd suggest moving the `pdfjs` import inside the `_pdfjs_version` function, as it's only needed there (via `version_info`).

Reply:
> I'd suggest moving the `pdfjs` import inside the `_pdfjs_version` function, as it's only needed there (via `version_info`).

Unfortunately there is another circular import issue,
```bash
$ python -m qutebrowser                                                           
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "qutebrowser/qutebrowser/__main__.py", line 29, in <module>
    sys.exit(qutebrowser.qutebrowser.main())
  File "qutebrowser/qutebrowser/qutebrowser.py", line 203, in main
    from qutebrowser import app
  File "qutebrowser/qutebrowser/app.py", line 54, in <module>
    from qutebrowser.commands import runners
  File "qutebrowser/qutebrowser/commands/runners.py", line 30, in <module>
    from qutebrowser.api import cmdutils
  File "qutebrowser/qutebrowser/api/cmdutils.py", line 55, in <module>
    from qutebrowser.utils import qtutils
  File "qutebrowser/qutebrowser/utils/qtutils.py", line 50, in <module>
    from qutebrowser.utils import usertypes, version
  File "qutebrowser/qutebrowser/utils/version.py", line 55, in <module>
    from qutebrowser.browser.webengine import webenginesettings
  File "qutebrowser/qutebrowser/browser/webengine/webenginesettings.py", line 35, in <module>
    from qutebrowser.browser.webengine import spell, webenginequtescheme
  File "qutebrowser/qutebrowser/browser/webengine/webenginequtescheme.py", line 31, in <module>
    from qutebrowser.browser import qutescheme
  File "qutebrowser/qutebrowser/browser/qutescheme.py", line 40, in <module>
    from qutebrowser.browser import pdfjs, downloads, history
  File "qutebrowser/qutebrowser/browser/pdfjs.py", line 27, in <module>
    from qutebrowser.utils import (utils, javascript, jinja, version, usertypes,
  File "qutebrowser/qutebrowser/utils/jinja.py", line 139, in <module>
    environment = Environment()
  File "qutebrowser/qutebrowser/utils/jinja.py", line 96, in __init__
    self.globals['qcolor_to_qsscolor'] = qtutils.qcolor_to_qsscolor
```
Similarly I could avoid this circular import issue by moving `webenginesettings` import inside `_chromium_version` but pytlint test would have the following complain,
```
tests/unit/utils/test_version.py:875:11: E1101: Module 'qutebrowser.utils.version' has no 'webenginesettings' member (no-member)
tests/unit/utils/test_version.py:877:32: E1101: Module 'qutebrowser.utils.version' has no 'webenginesettings' member (no-member)
tests/unit/utils/test_version.py:883:8: E1101: Module 'qutebrowser.utils.version' has no 'webenginesettings' member (no-member)
tests/unit/utils/test_version.py:894:8: E1101: Module 'qutebrowser.utils.version' has no 'webenginesettings' member (no-member)
tests/unit/utils/test_version.py:900:28: E1101: Module 'qutebrowser.utils.version' has no 'webenginesettings' member (no-member)
tests/unit/utils/test_version.py:990:7: E1101: Module 'qutebrowser.utils.version' has no 'webenginesettings' member (no-member)
tests/unit/utils/test_version.py:993:8: E1101: Module 'qutebrowser.utils.version' has no 'webenginesettings' member (no-member)
```

Reply:
Argh, this is a pain... :sweat: Even with that resolved, there's more:

```
ImportError while loading conftest '/home/florian/proj/qutebrowser/git/tests/conftest.py'.
tests/conftest.py:37: in <module>
    from helpers.messagemock import message_mock
tests/helpers/messagemock.py:27: in <module>
    from qutebrowser.utils import usertypes, message
qutebrowser/utils/usertypes.py:30: in <module>
    from qutebrowser.utils import log, qtutils, utils
qutebrowser/utils/qtutils.py:51: in <module>
    from qutebrowser.utils import usertypes, version
qutebrowser/utils/version.py:50: in <module>
    from qutebrowser.utils import log, utils, standarddir, usertypes, message
qutebrowser/utils/standarddir.py:34: in <module>
    from qutebrowser.utils import log, debug, message, utils
qutebrowser/utils/debug.py:34: in <module>
    from qutebrowser.utils import log, utils, qtutils, objreg
qutebrowser/utils/objreg.py:231: in <module>
    default: Any = usertypes.UNSET,
E   AttributeError: partially initialized module 'qutebrowser.utils.usertypes' has no attribute 'UNSET' (most likely due to a circular import)
```

Some day I should probably find out how to generate a big drawing of all dependencies between modules, print that out, and then find out how to untangle/reorganize everything nicely...

What about moving `version_check` to `utils.py` instead? I still think it'd better fit there (as it can be used for versions which have nothing to do with Qt), and that'd be the next best place after `version.py`.